### PR TITLE
implement some audit feedback

### PR DESCRIPTION
**Pull request by plaintextpaco**
This PR includes all of budi's ad-hoc, small changes, a linter run/small cleanup on AludelFactory's tests and some testcases for the addProgram method

I choose to submit this PR here since it's getting messy and will probably be a pain to review if I add much more to it.

Further improvements will be commited as separate PRs, such as #17 which aims to remove the InstanceRegistry.

### CODE REVIEW

**Comment by itirabasso** [comment]
@plaintextpaco I like how you to manage to use gherkin in solidity.

Thanks for running the linter, it has been delayed so long.

File: src/test/EnumerableSet.t.sol [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        templates.add(DEFAULT);
+
+        EnumerableSet.TemplateData memory value = templates.at(address(1337));
+
+        assertEq(value.template, address(1337));
+        assertEq(value.disabled, false);
+        assertEq(value.name, 'template');
+    }
+
+    // function
```
Comment (itirabasso) [inline src/test/EnumerableSet.t.sol:75]: delet this

File: src/test/EnumerableSet.t.sol [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    //         bool v = templates.add(EnumerableSet.TemplateData({
+    //             template: address(uint160(1337+i)),
+    //             name: string(abi.encode('template', i)),
+    //             disabled: false
+    //         }));
+    //         assertTrue(v);
+    //     }
+
+    //     assertEq(templates.length(), length);
+    // }
```
Comment (itirabasso) [inline src/test/EnumerableSet.t.sol:109]: This was commented for a reason. We should know why.

File: src/test/EnumerableSet.t.sol [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        assertTrue(templates.update(DEFAULT.template, true));
+
+        data = templates.at(DEFAULT.template);
+        assertEq(data.disabled, true);        
+    }
+
+    function test_update_nonexistent() public {
+        EnumerableSet.TemplateData memory data;
+
+        // this reverts as expected but i dont know how to test it
```
Comment (itirabasso) [inline src/test/EnumerableSet.t.sol:166]: Do we know how to test it now?

**Review by itirabasso — COMMENTED** [review]
Check what has been left undone and remove commented code.

File: src/test/EnumerableSet.t.sol (hunk 2 above)
Comment (plaintextpaco) [reply src/test/EnumerableSet.t.sol:109]: I wonder who we could ask about it:

image

Comment (plaintextpaco) [reply src/test/EnumerableSet.t.sol:109]: the next iteration of this testcase is implemented right above it, I'll remove this

File: src/contracts/AludelFactory.sol [hunk 4]
```diff
@@ -188,7 +191,9 @@ contract AludelFactory is Ownable, InstanceRegistry {
         onlyOwner
     {
         // check if the address is already registered
-        require(isInstance(program));
+        if(!isInstance(program)){
+          revert AludelNotRegistered();
+        }
```
Comment (itirabasso) [inline src/contracts/AludelFactory.sol:196]: yay custom errors

**Review by itirabasso — APPROVED** [review]

**Pull request merged**